### PR TITLE
Upgrade

### PR DESCRIPTION
**Pull request by yalh76**
## Problem

- *Level1*

## Solution

- *[Apply example_ynh](https://github.com/YunoHost-Apps/technitium-dns_ynh/commit/a81f54c4019aafde2d158b61e8d3cc54c0ab3919)*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by yalh76** [comment]
!testme

**Comment by yalh76** [comment]
!testme

**Comment by yalh76** [comment]
!testme

File: scripts/remove [hunk 1]
```diff
@@ -100,7 +92,10 @@ fi
 #=================================================
 # REMOVE VARIOUS FILES
 #=================================================
-ynh_script_progression --message="Removing various files..." --time --weight=1
+ynh_script_progression --message="Removing various files..." --weight=1
+
+ynh_secure_remove --file="/usr/bin/dotnet"
+ynh_secure_remove --file="/opt/dotnet"
```
Comment (navanchauhan) [inline scripts/remove:98]: will also need to change multi_instance to false then

**Review by navanchauhan — CHANGES_REQUESTED** [review]
LGTM except the secure removal of /opt/dotnet without checking if another instance is installed

File: scripts/remove (hunk 1 above)
Comment (yalh76) [reply scripts/remove:98]: it has been 

https://github.com/YunoHost-Apps/technitium-dns_ynh/blob/a2d9cb93351b22e65a575a9996e2f4d635ae7d80/manifest.json#L25

https://github.com/YunoHost-Apps/technitium-dns_ynh/blob/a2d9cb93351b22e65a575a9996e2f4d635ae7d80/check_process#L16

Comment (navanchauhan) [reply scripts/remove:98]: works for me 👍 

Thank you for your effort

**Review by navanchauhan — APPROVED** [review]

**Comment by yalh76** [comment]
> LGTM except the secure removal of /opt/dotnet without checking if another instance is installed

As far as I've understood, technitium-dns is listening on port 53

https://github.com/YunoHost-Apps/technitium-dns_ynh/blob/a2d9cb93351b22e65a575a9996e2f4d635ae7d80/scripts/install#L137

so it can't be multi-instance


By the maybe we need to open port 53 ?

**Comment by navanchauhan** [comment]
> > LGTM except the secure removal of /opt/dotnet without checking if another instance is installed
> 
> As far as I've understood, technitium-dns is listening on port 53
> 
> https://github.com/YunoHost-Apps/technitium-dns_ynh/blob/a2d9cb93351b22e65a575a9996e2f4d635ae7d80/scripts/install#L137
> 
> so it can't be multi-instance as only one service can listen on one port
> 
> @navanchauhan : By the maybe we need to open port 53 and close it during removal? I don't think `--needs_exposed_ports` open and close ports ...

Port 53 is already opened because of dnsmasq

**Comment by yalh76** [comment]
fine then ^^

**Pull request merged**